### PR TITLE
Feature: CORE -> calcular aportes y deudas

### LINKED ISSUE #92: CORE - Calcular aportes y deudas
Puro. Deterministico. SOLIDO.

- [x] Pensarlo - 2
- [x] Documentarlo - 1
- [x] Hacerlo - 2
- [x] Testearlo - 1

Routes
- R (any) Leer Aportes
- R (any) Leer Deudas

Image

### PR DESCRIPTION
**Pull request by Facugarciar0805**
This pull request introduces a new core balance calculation engine for group transactions, along with a comprehensive test suite and a new `user_wallet` model. The main focus is on accurately tracking and updating user and group balances in response to deposits, withdrawals, and expenses, with robust error handling and support for high-precision decimal arithmetic. The changes are organized into new modules, and extensive tests ensure correctness and edge case coverage.

**Core balance calculation engine:**

- Introduced the `BalancesMap` struct in `balances_map.rs`, which manages per-user and group balances using high-precision arithmetic. It provides methods for adding balance to a user, distributing withdrawals, and querying balances, with error handling for non-existent users and insufficient funds.
- Added the `core` function in `core.rs` to reconstruct the current state of all balances based on a list of users, transactions, and expenses. This function enforces data integrity by panicking if transactions reference unknown users or if withdrawals exceed available funds.
- Integrated the new core modules into the project structure (`mod.rs`, `main.rs`). [[1]](diffhunk://#diff-7f67b2dbac45d35ca025e1037b855083c7ca85a8b6be03ee24243aa4681af496R1-R3) [[2]](diffhunk://#diff-01fd5002f7585d6f2b853a8c83e6c0a1acb2b16b6e6f8f581fb3365377d331bcR4)

**Testing and validation:**

- Added a comprehensive test suite for the balance engine, including unit tests for `BalancesMap` (covering initialization, balance updates, error conditions, and edge cases like rounding), as well as integration tests for transaction and expense scenarios. [[1]](diffhunk://#diff-9a875843e0f8be1ac8c74dd395d386d21a608a8f97cd2ea5dd74a5870e167755R1-R170) [[2]](diffhunk://#diff-e7c1fc5465cde83154a81f98052c4c7f1ab71bec52fb2af8670a07896f0d0b76R1-R53) [[3]](diffhunk://#diff-da2ad8c2c1161440c74665181315386b8d1fcc21fbb3b5fe88175cf4a17383c2R1-R79) [[4]](diffhunk://#diff-b075eb823a31ee232ceacd9b5667ab59a4d81082c6a87b7063bcbdebb00ca721R1-R63) [[5]](diffhunk://#diff-d4af4229e02bd2a902c6817b423d3b6b6335e31d30e514cba3e7b89ebde2a5e3R1-R53) [[6]](diffhunk://#diff-821c532dc48536f9fbdd4ddec22629d14b54612d5dd4819b5ebb9cc71dad5da1R1-R4) [[7]](diffhunk://#diff-6de6ebf34754e1ff1a837308c87eacabd79b7e0b5b308343ed03be3e2d35e033R1) [[8]](diffhunk://#diff-37be95672277084e13e149820b930178de5a4997ce20f24ac7a7aeb3ef274a4eR1-R2)
- Implemented test helpers for generating dummy data and simplifying test code.

**User wallet model:**

- Added a new `user_wallet` module and model, including database mappings and serialization structures for handling user wallet addresses, balances, and associated currencies. [[1]](diffhunk://#diff-dd0b5157ebbf03b6656be546366ec087f0ee1220ae2ab11544d9370f9234bd77R9) [[2]](diffhunk://#diff-8eceee1cd21da1678889c368a3f4748f465f6c6c35930cbcbdaca0ad200fd574R1-R37)

These changes provide a robust, extensible foundation for group-based financial operations, with strong guarantees of correctness and clear extensibility for future features such as investments.

### CODE REVIEW

File: server/src/models/user_wallet.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+#[derive(Queryable, Serialize)]
+pub struct WalletWithTickerDb {
+    pub wallet_id: Uuid,
+    pub address: String,
+    pub balance: BigDecimal,
+    pub currency_id: Uuid,
+    pub ticker: String,
+}
```
[the remark this answers is not in the document]
Comment (MateoDi9z) [reply server/src/models/user_wallet.rs:37]: Esto ya esta fixeado en la branch, si te haces un merge ?

**Comment by MateoDi9z** [comment]
Lo movi a otra PR porque esta se rompio
